smartsim/_core/generation/modelwriter.py

```diff
@@ -116,8 +116,8 @@ def _replace_tags(
         :returns: A dict of parameter names and values set for the file
         """
         edited = []
-        unused_tags: t.DefaultDict[str, t.List[int]] = collections.defaultdict(list)
-        used_params: t.Dict[str, str] = {}
+        unused_tags: defaultdict[str, list[int]] = collections.defaultdict(list)
```

Comment:
Nit: Now that we are explicitly importing `defaultdict` into the namespace there is no need to call the constructor off the `collections` module anymore
```suggestion
        unused_tags = defaultdict[str, list[int]](list)
```